pkg/routes/openidmetadata.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package routes
+
+import (
+	"net/http"
+
+	restful "github.com/emicklei/go-restful"
+
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/serviceaccount"
+)
+
+// This code is in package routes because many controllers import
+// pkg/serviceaccount, but are not allowed by import-boss to depend on
+// go-restful. All logic that deals with keys is kept in pkg/serviceaccount,
+// and only the rendered JSON is passed into this server.
+
+const (
+	// cacheControl is the value of the Cache-Control header. Overrides the
+	// global `private, no-cache` setting.
+	headerCacheControl = "Cache-Control"
+	cacheControl       = "public, max-age=86400" // 1 day
```

Comment:
I don't know what best-practice is for this, but a day seems long. 

for reference:
```sh
$ curl -v https://accounts.google.com/.well-known/openid-configuration 2>&1 | grep cache
< cache-control: public, max-age=3600
```

Reply:
Updated to an hour per the accounts.google.com example.